Ds3/Helpers/Transferrers/WriteTransferrer.cs

```diff
@@ -43,7 +37,26 @@ internal class WriteTransferrer : ITransferrer
                 request.WithMetadata(metadataAccess.GetMetadataValue(objectName));
             }
 
-            client.PutObject(request);
+            try
+            {
+                client.PutObject(request);
+            }
+            catch (Exception ex)
+            {
+                if (retransmitRetries == 0)
+                {
+                    throw new Ds3NoMoreRetransmitException(
+                        string.Format(Resources.NoMoreRetransmitException, request.ObjectName, request.Offset.Value),
+                        ex);
+                }
+
+                if (!stream.CanSeek) throw new Ds3NotSupportedStream(Resources.NotSupportedStream, ex);
+
+                retransmitRetries--;
+                stream.Position = request.Offset.Value;
+                Transfer(client, bucketName, objectName, blobOffset, jobId, ranges, stream, metadataAccess,
```

Comment:
Are you using recursion to do the retry logic?  If this is the case, this should be updated to use a loop instead.